src/tracker/long_tracker.jl

```diff
@@ -0,0 +1,161 @@
+    trajectories.ID = trajectories.uuid
+    # list the uuid in the leftmost column
+    cols = [col for col in names(trajectories) if col ∉ ["ID", "uuid"]]
+    return trajectories[!, ["ID", cols...]]
+end
+
+"""
+    find_floe_matches(
+    tracked,
+    candidate_props,
+    condition_thresholds,
+    mc_thresholds
+)
+
+Find matches for floes in `tracked` from floes in  `candidate_props`.
+
+# Arguments
+- `tracked`: dataframe containing floe trajectories.
+- `candidate_props`: dataframe containing floe candidate properties.
+- `condition_thresholds`: thresholds for deciding whether to match floe `i` from tracked to floe j from `candidate_props`
+- `mc_thresholds`: thresholds for area mismatch and psi-s shape correlation
+"""
+function find_floe_matches(
+    tracked::T,
+    candidate_props::T,
+    condition_thresholds,
+    mc_thresholds
+) where {T<:AbstractDataFrame}
+    props1 = deepcopy(tracked)
+    props2 = deepcopy(candidate_props)
+    match_total = MatchedPairs(props2)
+    while true # there are no more floes to match in props1
+
+        # This routine mutates both props1 and props2.
+        # Get preliminary matches for floe r in props1 from floes in props2
+
+        # Container for props of matched floe pairs and their similarity ratios. Matches will be updated and added to match_total
+        matched_pairs = MatchedPairs(props2)
+        for r in 1:nrow(props1) # TODO: consider using eachrow(props1) to iterate over rows
+            # 1. Collect preliminary matches for floe r in matching_floes
+            matching_floes = makeemptydffrom(props2)
+
+            for s in 1:nrow(props2) # TODO: consider using eachrow(props2) to iterate over rows
+                Δt = get_dt(props1, r, props2, s)
+                @debug "Considering floe 2:$s for floe 1:$r"
+                ratios, conditions, dist = compute_ratios_conditions(
+                    (props1, r), (props2, s), Δt, condition_thresholds
+                )
+
+                if callmatchcorr(conditions)
+                    @debug "Getting mismatch and correlation for floe 1:$r and floe 2:$s"
+                    (area_mismatch, corr) = matchcorr(
+                        props1.mask[r], props2.mask[s], Δt; mc_thresholds.comp...
+                    )
+
+                    if isfloegoodmatch(
+                        conditions, mc_thresholds.goodness, area_mismatch, corr
+                    )
+                        @debug "** Found a good match for floe 1:$r => 2:$s"
+                        appendrows!(
+                            matching_floes,
+                            props2[s, :],
+                            (ratios..., area_mismatch, corr),
+                            s,
+                            dist,
+                        )
+                        @debug "Matching floes" matching_floes
+                    end
+                end
+            end # of s for loop
+
+            # 2. Find the best match for floe r
```

Comment:
Do you need to add a check in here that you don't have multiple matches?

Reply:
This is how the algorithm is implemented in the original MATLAB script. During the first phase of pairing there could be multiple matches. `resolvecollisions!` is supposed to handle that situation.